src/lib/storage/chunk.hpp

```diff
@@ -206,6 +207,10 @@ class Chunk : private Noncopyable {
 
   const PolymorphicAllocator<Chunk>& get_allocator() const;
 
+  std::shared_ptr<ChunkStatistics> statistics() const { return _statistics; }
+
+  void set_statistics(std::shared_ptr<ChunkStatistics> statistics) { _statistics = statistics; }
```

Comment:
Consider debugasserting that the statistics object has the correct number of columns.
If you want to make me happy, also debugassert that the column types match.

Reply:
i dont think there is an easy way to check if the column types match, how would we get the information in this context? correct number of columns is good tho :+1:

Reply:
you're right, as of the state of this PR that's not easily possible because the chunk doesn't know the type of his columns.